retrieval/target.go

```diff
@@ -212,12 +211,15 @@ func (t *target) Update(cfg *config.ScrapeConfig, baseLabels clientmodel.LabelSe
 	}
 }
 
-func (t *target) String() string {
+func (t *Target) String() string {
 	return t.url.Host
 }
 
-// Ingest implements Target and extraction.Ingester.
-func (t *target) Ingest(s clientmodel.Samples) error {
+// Ingest implements an extraction.Ingester.
+func (t *Target) Ingest(s clientmodel.Samples) error {
+	t.RLock()
+	deadline := t.deadline
```

Comment:
`deadline` is the scrape timeout defined in the config. Thus it can be changed any time. This happens in `Target.Update`.
We no longer lock the target during scraping and updates can thus happen anytime - basically the whole reason for this PR.

One could argue about copying the deadline initially before starting to scrape but I see no harm if a new deadline becomes effective during the scrape.